src/compiler/checker.ts

```diff
@@ -15082,6 +15089,9 @@ namespace ts {
                         break;
 
                     case SyntaxKind.StaticKeyword:
+                        if (node.kind === SyntaxKind.Identifier && reservedStaticPropertyNames[node.symbol.name]) {

```

Comment:
this makes static declarations with name "toString" for instance illegal, if you are using a map, use `ts.hasProperty` to check if the property exist on the object it self and not its prototype chain.